src/software/ai/hl/stp/tactic/dribble/dribble_fsm.h

```diff
@@ -0,0 +1,304 @@
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), intercept_position, face_ball_orientation, 0,
+                DribblerMode::MAX_FORCE, BallCollisionType::ALLOW,
+                AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        /**
+         * Action to dribble the ball
+         *
+         * This action will orient the robot towards the destination, dribble to the
+         * destination, and then pivot to face the expected orientation
+         *
+         * @param event DribbleFSM::Update
+         */
+        const auto dribble = [this, &continuous_dribbling_start_point](auto event) {
+            Point ball_position = event.common.world.ball().position();
+            auto [target_destination, target_orientation] =
+                calculateNextDribbleDestinationAndOrientation(
+                    event.common.world.ball(), event.common.robot,
+                    event.control_params.dribble_destination,
+                    event.control_params.final_dribble_orientation);
+            AutoChipOrKick auto_chip_or_kick = AutoChipOrKick{AutoChipOrKickMode::OFF, 0};
+
+            if (!event.control_params.allow_excessive_dribbling &&
+                !comparePoints(ball_position, continuous_dribbling_start_point,
+                               MAX_CONTINUOUS_DRIBBLING_DISTANCE))
+            {
+                // give the ball a little kick
+                auto_chip_or_kick =
+                    AutoChipOrKick{AutoChipOrKickMode::AUTOKICK, DRIBBLE_KICK_SPEED};
+            }
+
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), target_destination, target_orientation, 0,
+                DribblerMode::MAX_FORCE, BallCollisionType::ALLOW, auto_chip_or_kick,
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        const auto start_dribble = [this, &continuous_dribbling_start_point,
+                                    dribble](auto event) {
+            // update continuous_dribbling_start_point once we start dribbling
+            continuous_dribbling_start_point = event.common.world.ball().position();
+            dribble(event);
+        };
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *get_possession_s + update_e[have_possession] / start_dribble = dribble_s,
+            get_possession_s + update_e[!have_possession] / get_possession,
+            dribble_s + update_e[!have_possession] / get_possession = get_possession_s,
+            dribble_s + update_e[!dribbling_done] / dribble,
+            dribble_s + update_e[dribbling_done] / dribble  = X,
+            X + update_e[!have_possession] / get_possession = get_possession_s,
+            X + update_e[!dribbling_done] / dribble = dribble_s, X + update_e / dribble);
```

Comment:
What is the last transition doing? `X + update_e / dribble`

Reply:
it's continuously calling the dribble action from the terminal state and doing a self transition back to the terminal state